src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -154,36 +155,27 @@ private void processDirectors() {
                 if (p == null || p.getInventory().getItemInMainHand().getType() != Config.DirectorTool) {
                     continue;
                 }
-                Vector tv = tnt.getVelocity();
-                double speed = tv.length(); // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
-                tv = tv.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
+                Vector tntVector = tnt.getVelocity();
+                // Store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
+                // We're only interested in the horizontal speed for now since that's all directors *should* affect.
+                double horizontalSpeed = new Vector(tntVector.getX(),0,tntVector.getY()).length();
+                tntVector = tntVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
```

Comment:
This normalize still takes into account the vertical vector.  I'd instead create the tntVector using the method you used for horizantalSpeed.

Reply:
doesn't the `tntVector.setY(0) account for the vertical vector?`

Reply:
oh, wait, for some reason it seems to still be showing an old version of this file

Reply:
I don't see a setY anywhere but the end